### PR TITLE
Added update command to tool.sh and also ignored joe/data

### PR DESCRIPTION
**Pull request by syndbg**
Just a handy thing if we want to have the latest gitignore files.

However the command name is debatable. For convenience I kept it short, but update can mean a lot of things.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,9 @@ __pycache__/
 # C extensions
 *.so
 
+# .gitignore repo
+joe/data
```
Comment (karan) [inline .gitignore:9]: Nope. This won't work.

If we do not package `joe/data/` files, the end user will have to download the files manually, which is extremely counter-intuitive.

File: tool.sh [hunk 2]
```diff
@@ -46,6 +47,8 @@ elif [ "$1" == "prod" ]; then
   readme
   # build and upload package to prod pypi
   python setup.py sdist upload -r pypi
+elif [ "$1" == "update" ]; then
+  git submodule foreach git pull origin master
```
Comment (karan) [inline tool.sh:51]: What if we add more submodules in the future in different location? Do we want `update` to fetch all of them?

**Comment by karan** [comment]
I think it's a handy command, but it doesn't seem like it will be used very much. The original repo is updated quite often but we don't want to push updates to the whole tool in that frequency. Best to treat it like a "private command" that can only be run by those with permissions to push updates to the whole tool.

File: tool.sh (hunk 2 above)
Comment (syndbg) [inline tool.sh:51]: I think so. If we use other submodules I don't see how will we hurt from newest versions. 

But what do you have in mind?

File: .gitignore (hunk 1 above)
Comment (syndbg) [inline .gitignore:9]: I'm not  sure about this, but
if we already have the files in our repo and put `joe/data` in .gitignore, the `data` will remain but the folder won't be tracked in the future.

If the above behavior is true, isn't that favorable for our situation?

**Pull request closed without merging**